### PR TITLE
Ndlano/issues#1043 add frontpage api

### PR DESCRIPTION
**Pull request by chrpeter**

### CODE REVIEW

File: src/schema.ts [hunk 1]
```diff
@@ -147,23 +147,56 @@ type Topic {
   supplementaryResources: [Resource]
 }
 
+type FrontpageSubjects {
+  category: String
+  subjects: [String]
```
Comment (oyvinmar) [inline src/schema.ts]: This should return [Subjects]. So we can have a query like this:
```gql
{
  frontpage {
    categories {
      name 
      subjects {
        name
      }
    }
  }
}
```
 Make sure subjects is fetched with a loader.

File: src/schema.ts [hunk 2]
```diff
@@ -147,23 +147,56 @@ type Topic {
...
+}
+
+type Frontpage {
+  topical: [String]
+  subjects: [FrontpageSubjects]
+}
+
+type SubjectPageArticles {
+  location: String
+  articleIds: [String]
```
Comment (oyvinmar) [inline src/schema.ts]: Should be renamed to to resources and return `[Resource]`. Resource type should be expanded with `meta: ArticleSubset`

File: src/schema.ts [hunk 3]
```diff
@@ -147,23 +147,56 @@ type Topic {
...
+}
+
+type SubjectPageTopical {
+  location: String
+  id: String
```
Comment (oyvinmar) [inline src/schema.ts]: Should probably be `resource: Resource`

File: src/schema.ts [hunk 4]
```diff
@@ -30,6 +30,7 @@ type Resource {
   path: String
   article: Article
   resourceTypes: [ResourceType]
+  meta: ArticleSubset
```
Comment (oyvinmar) [inline src/schema.ts]: Need to expand ArticleSubset with metaImage when https://github.com/NDLANO/Issues/issues/1052 is ready.

File: src/schema.ts [hunk 5]
```diff
@@ -23,13 +23,23 @@ type ResourceType {
...
+}
+
 type Resource {
   id: String!
   name: String!
   contentUri: String
   path: String
-  article: Article
   resourceTypes: [ResourceType]
+  meta: ResourceMeta
```
Comment (oyvinmar) [inline src/schema.ts]: Shouldn't this be ArticleSubset?

Comment (oyvinmar) [reply src/schema.ts]: Or ArticleSubset renamed to ResourceMeta or just Meta.

File: src/data/api.ts [hunk 6]
```diff
@@ -149,3 +149,63 @@ export async function fetchArticles(
...
+      : undefined,
+    lastUpdated: learningPath.lastUpdated,
+  };
+}
+
+export async function fetchArticleMeta(
+  articleId: string,
+  context: Context,
+): Promise<GQLResourceMeta> {
+  const response = await fetch(
```
Comment (oyvinmar) [inline src/data/api.ts]: Don't call article-converter here, use article-api if you don't need the article content. Should probably use articlesLoader in the resolvers if possible.

**Comment by chrpeter** [comment]
@oyvinmar Should have refactored it now.

File: src/resolvers.ts [hunk 7]
```diff
@@ -152,6 +215,34 @@ export const resolvers = {
...
+        resource.contentUri &&
+        resource.contentUri.startsWith('urn:article')
+      ) {
+        return context.loaders.articlesLoader.load(
+          resource.contentUri.replace('urn:article:', ''),
+        );
+      }
+      throw Object.assign(
+        new Error(
+          'Missing learningpath contentUri for resource with id: ' +
```
Comment (oyvinmar) [inline src/resolvers.ts]: Remove learningpath from error message.

**Pull request merged**